### PR TITLE
refactor(generator): split translator vs. parser

### PR DESCRIPTION
**Pull request by coryan**
This change introduces a new interface `Parser`. Its job is to convert the
textual representation of an API specification (i.e., the OpenAPI or Protobuf
inputs) into the `genclient.API` AST.

In this PR we create the parser and split the Protobuf and OpenAPI translators
to depend on them. A future PR will create a single standalone program to call
the parser and per-language Codec.

Part of the work for #150

### CODE REVIEW

File: generator/internal/genclient/parser/openapi/openapi.go [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+	document, err := libopenapi.NewDocument(contents)
+	if err != nil {
+		return nil, err
+	}
+	docModel, errors := document.BuildV3Model()
+	if len(errors) > 0 {
+		for i := range errors {
+			fmt.Printf("error: %e\n", errors[i])
+		}
+		return nil, fmt.Errorf("cannot convert document to OpenAPI V3 model: %e", errors[0])
```
Comment (codyoss) [inline generator/internal/genclient/parser/openapi/openapi.go:82]: remove printing above and do:

```suggestion
		return nil, fmt.Errorf("cannot convert document to OpenAPI V3 model: %w", errors.Join(errs...))
```

File: generator/internal/genclient/parser/openapi/openapi.go [hunk 2]
```diff
@@ -0,0 +1,656 @@
...
+	// Translates OpenAPI specification into a [genclient.GenerateRequest].
+	return makeAPI(serviceConfig, model)
+}
+
+func createDocModel(contents []byte) (*libopenapi.DocumentModel[v3.Document], error) {
+	document, err := libopenapi.NewDocument(contents)
+	if err != nil {
+		return nil, err
+	}
+	docModel, errors := document.BuildV3Model()
```
Comment (codyoss) [inline generator/internal/genclient/parser/openapi/openapi.go:77]:
```suggestion
	docModel, errs := document.BuildV3Model()
```

File: generator/internal/genclient/parser/openapi/openapi.go [hunk 3]
```diff
@@ -0,0 +1,656 @@
...
+	if len(errors) > 0 {
```
Comment (codyoss) [inline generator/internal/genclient/parser/openapi/openapi.go:78]:
```suggestion
	if len(errs) > 0 {
```

File: generator/internal/genclient/parser/parser.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	return map[string]newParser{
+		"openapi":  func() genclient.Parser { return openapi.NewParser() },
+		"protobuf": func() genclient.Parser { return protobuf.NewParser() },
+	}
+}
+
+func NewTranslator(parserID string) (genclient.Parser, error) {
+	create, ok := knownParsers()[parserID]
+	if !ok {
+		return nil, fmt.Errorf("unknown parser (%s)", parserID)
```
Comment (codyoss) [inline generator/internal/genclient/parser/parser.go:38]: nit:
```suggestion
		return nil, fmt.Errorf("unknown parser %q", parserID)
```

File: generator/internal/genclient/parser/parser.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+type newParser func() genclient.Parser
+
+func knownParsers() map[string]newParser {
+	return map[string]newParser{
+		"openapi":  func() genclient.Parser { return openapi.NewParser() },
+		"protobuf": func() genclient.Parser { return protobuf.NewParser() },
+	}
+}
+
+func NewTranslator(parserID string) (genclient.Parser, error) {
```
Comment (codyoss) [inline generator/internal/genclient/parser/parser.go:35]:
```suggestion
func NewParser(parserID string) (genclient.Parser, error) {
```

**Review by codyoss — APPROVED** [review]
Mostly just nits

File: generator/internal/genclient/parser/parser.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	create, ok := knownParsers()[parserID]
+	if !ok {
+		return nil, fmt.Errorf("unknown parser (%s)", parserID)
+	}
+	return create(), nil
+}
+
+func Help() string {
```
[the remark this answers is not in the document]
Comment (coryan) [reply generator/internal/genclient/parser/parser.go:43]: Not yet. I have plans for this to appear in a command-line program.  If you don't mind, let's leave it, we can remove it later if it was a bad idea.

File: generator/internal/genclient/parser/openapi/openapi.go (hunk 2 above)
Comment (coryan) [reply generator/internal/genclient/parser/openapi/openapi.go]: Done.

File: generator/internal/genclient/parser/openapi/openapi.go (hunk 3 above)
Comment (coryan) [reply generator/internal/genclient/parser/openapi/openapi.go]: Done.

File: generator/internal/genclient/parser/openapi/openapi.go (hunk 1 above)
Comment (coryan) [reply generator/internal/genclient/parser/openapi/openapi.go]: Done. TIL.

File: generator/internal/genclient/parser/parser.go (hunk 5 above)
Comment (coryan) [reply generator/internal/genclient/parser/parser.go]: Done.

File: generator/internal/genclient/parser/parser.go (hunk 4 above)
Comment (coryan) [reply generator/internal/genclient/parser/parser.go]: Done.

**Pull request merged**